qt/applications/workbench/workbench/plugins/editor.py

```diff
@@ -73,6 +76,22 @@ def app_closing(self):
         """
         return self.editors.close_all()
 
+    def dragEnterEvent(self, event):
+        data = event.mimeData()
+        if data.hasText() and data.hasUrls:
+            filepaths = [url.toLocalFile() for url in data.urls()]
+            for filepath in filepaths:
+                if filepath.endswith('.py') or filepath.endswith('.pyw'):
+                    event.acceptProposedAction()
+
+    def dropEvent(self, event):
+        data = event.mimeData()
+        for url in data.urls():
+            filepath = url.toLocalFile()
+            if filepath.endswith('.py') or filepath.endswith('.pyw'):
+                if osp.isfile(filepath):
+                    self.open_file_in_new_tab(filepath)
+
```

Comment:
Perhaps emit a warning if a file could not be opened so the user knows something has happened?